rest/route/github.go

```diff
@@ -558,7 +559,12 @@ func (gh *githubHookApi) refreshPatchStatus(ctx context.Context, owner, repo str
 	return nil
 }
 
-func (gh *githubHookApi) AddIntentForPR(pr *github.PullRequest, owner, calledBy string) error {
+// AddIntentForPR takes a pr and checks if that pr is applicable for CI and if it is, does one of two things.
+// If overrideExisting is false, it only creates a PR if no PR has the same head sha. If any exist, it
```

Comment:
This comment overall looks good (and thanks for adding a comment!), but a brief explanation of what the head SHA is would help clarify what the issue is that we're trying to solve/why it can happen so we understand this special case later on.